### PR TITLE
Feature/eclair improvements

### PR DESCRIPTION
**Pull request by kiwiidb**
### Describe the changes you have made in this PR

- Add `/listinvoices` function to connector to avoid error messages. However, Eclair has currently no way of listing only settled invoices, so all created invoices are displayed.
- Fixes a bug where a payment fee was displayed as 1000x too much.
- Use the [`/usablebalances`](https://acinq.github.io/eclair/#usablebalances) endpoint to fetch the balance, which is easier and more accurate than using the `/channels` endpoint. 

### Type of change

- `fix`: Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

Tested locally by using Eclair.

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Review by escapedcat — APPROVED** [review]
utack

File: src/extension/background-script/connectors/eclair.ts [hunk 1]
```diff
@@ -61,30 +69,38 @@ class Eclair implements Connector {
...
+        }) => ({
+          id: invoice.paymentHash,
+          memo: invoice.description,
+          settled: true,
+          settleDate: invoice.timestamp * 1000,
+          totalAmount: invoice.amount / 1000,
+          type: "received",
+        })
+      )
+      .sort((a: ConnectorInvoice, b: ConnectorInvoice) => {
```
Comment (reneaaron) [inline src/extension/background-script/connectors/eclair.ts:90]: Should we add some filtering then here (like we do for commando?):

https://github.com/getAlby/lightning-browser-extension/blob/master/src/extension/background-script/connectors/commando.ts#L206

Comment (reneaaron) [reply src/extension/background-script/connectors/eclair.ts:90]: I think the API should always return the same kind of data, also across connectors.

Comment (kiwiidb) [reply src/extension/background-script/connectors/eclair.ts:90]: Eclair doesn't support this kind of filter at the moment.

Comment (reneaaron) [reply src/extension/background-script/connectors/eclair.ts:90]: Exactly, the question is whether we should filter it on the client?

Comment (kiwiidb) [reply src/extension/background-script/connectors/eclair.ts:90]: This call doesn't even tell you if an invoice was settled or not :) 
The alternative is to use the [audit](https://acinq.github.io/eclair/#audit) api endpoint, but that misses the memo field.

Also if we are being realistic, there are very few people that use their Eclair node in this way so I wouldn't bother with too much more effort. In fact, we haven't received  a single complaint in the 1year+ that it was in the extension while it was broken.

Comment (reneaaron) [reply src/extension/background-script/connectors/eclair.ts:90]: > This call doesn't even tell you if an invoice was settled or not :)

Oh, okay then. 👍

**Review by reneaaron — APPROVED** [review]
See comment

utACK

File: src/extension/background-script/connectors/eclair.ts [hunk 2]
```diff
@@ -38,7 +39,14 @@ class Eclair implements Connector {
...
 
   get supportedMethods() {
-    return ["getInfo", "keysend", "makeInvoice", "sendPayment", "signMessage"];
+    return [
+      "getInfo",
+      "keysend",
+      "makeInvoice",
+      "sendPayment",
+      "signMessage",
+      "listInvoices",
```
[the remark this answers is not in the document]
Comment (kiwiidb) [reply src/extension/background-script/connectors/eclair.ts:48]: The point of this PR is not to add extra features but just to fix a few broken things. So I don't think it's necessary.

**Pull request merged**